sys/arm64/vmm/vmm_arm64.c

```diff
@@ -1189,6 +1189,9 @@ vmmops_run(void *vcpui, uintcap_t pc, pmap_t pmap, struct vm_eventinfo *evinfo)
 			/* Set the new cpsr */
 			hypctx->tf.tf_spsr = hypctx->spsr_el1 & PSR_FLAGS;
 			hypctx->tf.tf_spsr |= PSR_DAIF | PSR_M_EL1h;
+#if __has_feature(capabilities)
+			hypctx->tf.tf_spsr |= hypctx->spsr_el1 & PSR_C64;
```

Comment:
This would break a hybrid kernel if you happen to inject an exception while it's in a C64 region (e.g. copyin/out temporarily enable C64). You should respect the configured C64E instead:
```suggestion
			if ((hypctx->cctlr_el1 & CCTLR_EL1_C64E_MASK) != 0)
				hypctx->tf.tf_spsr |= PSR_C64;
```

Reply:
Technically this should also ignore C64E if CPACR_EL1.CEN[0] is 0, although in practice no guest will configure that as it's a pointless configuration.

Reply:
(Perhaps that should get added at the same time as fixing VBAR_EL1 to honour CEN)